transport/src/main/java/io/el/channel/AbstractChannelHandlerContext.java

```diff
@@ -0,0 +1,117 @@
+package io.el.channel;
+
+import static io.el.channel.ChannelHandlerFlag.FLAG_OUTBOUND;
+import static io.el.channel.ChannelHandlerFlag.flag;
+
+import io.el.concurrent.EventLoop;
+import io.el.internal.ObjectUtil;
+
+abstract public class AbstractChannelHandlerContext implements ChannelHandlerContext {
+
+  /**
+   * {@link ChannelHandlerContext} of the next {@link ChannelHandler} contained in the {@link ChannelPipeline}
+   * */
+  volatile AbstractChannelHandlerContext next;
```

Comment:
How does the `volatile` change the behavior?

Reply:
@majecty 
It is because other handlers can change the context state. So by using the `volatile` keyword, we can save the latest state to the main memory not in cache. So other handlers can read the same state.

Reply:
Thanks. I understood it. The `next` and `prev` variable can be changed from other thread at uncertain time!